src/hexadecimal.jl

```diff
@@ -0,0 +1,160 @@
+# All bytes that are not hex digits are 0x0000FFFF
+const _HEX_LUT = UInt32[
+    b in UInt8('0'):UInt8('9') ? UInt32(b - UInt8('0')) :
+        b in UInt8('a'):UInt8('f') ? UInt32(b - UInt8('a') + 0x0a) :
+        b in UInt8('A'):UInt8('F') ? UInt32(b - UInt8('A') + 0x0a) :
+        0xFFFF
+    for b
+    in UInt8(0):UInt8(255)
+]
+
+### SHA1 ###
+
+struct SHA1
```

Comment:
Do we expect this `SHA1` type to be a public user interface? For example, would CSV return a SHA1 column? (or would we expect CSV to define it's own type that represents SHA1?)

I think we should probably add a docstring. 

What i'm wondering is if we need to do to make this a reasonably usable type for downstream users to receive.  e.g. I'm wondering if we also define certain common functions (e.g. [that `UUID` suuports](https://github.com/JuliaLang/julia/blob/master/base/uuid.jl)). 

Like `string`/`print`/`show`: should `string(SHA1((1,2,3,4,5))) == "0000000100000002000000030000000400000005"`? Should we print SHA1 as `SHA1("0000000100000002000000030000000400000005")`?

Should we add constructors and convert methods? e.g. `SHA1(::String)`? or `SHA1(::Vector{UInt8})` (so we can construct a `SHA1` from `SHA.sha1`?

Also i think we'd need `isless` (so we can compare `SHA1`s) and `broadcastable` (so we can operate on a column of `SHA1`s)

Reply:
So I thought about this and I agree, that we want the `string` and `broadcastable`, but additional constructors are a little tricky, because people could expect this to actually calculate the sha1 hash of the input string or bytes. Since this is a storage-only type, I think it also make sense to "show it as you store it" for the `show` method.

Reply:
i think my preference would be to match UUID

But i'm happy to leave this as a topic for potential follow-up PRs